internal/k8s/env.go

```diff
@@ -25,40 +22,17 @@ func (e Env) IsLocalCluster() bool {
 	return e == EnvMinikube || e == EnvDockerDesktop || e == EnvMicroK8s
 }
 
-// If something goes wrong getting k8s Env, we want to hold onto err in case we need to
-// surface for debugging, but it might be an expected case so we don't want to stop wiring.
-type EnvOrError struct {
-	Env Env
-	Err error
+func ProvideEnv(kubeContext KubeContext) Env {
+	return EnvFromString(string(kubeContext))
 }
 
-func ProvideEnvOrError(ctx context.Context) EnvOrError {
-	kubeContext, err := detectKubeContext()
+func ProvideKubeContext(clientLoader clientcmd.ClientConfig) (KubeContext, error) {
```

Comment:
if you get rid of the `EnvOrError` struct and just have this return an error if it fails, then the whole wire-up will fail if we can't get kubeContext -- which is NOT what we want if the user is just using Docker Compose. Can you put back the envOrError so things don't blow up if you don't have kubernetes configured? (Or is that a case we don't actually want to support?)

Reply:
don't panic! it still works. now you get an error that looks like this:

```
Error: Error watching services. Are you connected to kubernetes?
: could not set up k8s client: invalid configuration: no configuration has been provided
```